### PR TITLE
feat(icu): add optional TextEngine plugin interface backed by ICU4X (Phase 1 — line breaking)

### PR DESCRIPTION
**Pull request by Vizards**
## Summary

This PR introduces an optional `TextEngine` plugin interface to satori's text processing pipeline, along with a companion `icu4satori` package that implements it using [ICU4X](https://github.com/unicode-org/icu4x).

The motivation and full roadmap are described in the tracking issue: https://github.com/vercel/satori/issues/743

**Key design principle: zero breaking changes.** The `textEngine` option is entirely opt-in. Any user who does not pass it sees no behavior change — the existing `linebreak` code path is untouched.

#### satori core (7 files)

| File | Change |
|------|--------|
| `src/satori.ts` | Add `TextEngine` interface definition; add optional `textEngine?` field to `SatoriOptions` |
| `src/layout.ts` | Thread `textEngine?` through `LayoutContext` |
| `src/text/index.ts` | Extract `textEngine` from `context` and pass it to `preprocess()` |
| `src/text/processor.ts` | Add `lineBreak` and `textEngine` parameters to `preprocess()` / `processWordBreak()` |
| `src/utils.ts` | Add ICU4X code path to `splitByBreakOpportunities()`; original `linebreak` path is unchanged when `textEngine` is not provided |
| `src/handler/expand.ts` | Add `lineBreak` field to `MainStyle` |
| `src/handler/inheritable.ts` | Add `lineBreak` to the set of inheritable CSS properties (fixing missing CSS inheritance behavior) |

#### icu4satori package (new, `packages/icu4satori/`)

- `src/index.ts`: exports `init()`, `createTextEngine()`, and the `TextEngine` interface
- `src/line-break.ts`: implements `getLineBreaks()` using ICU4X `LineSegmenter` + `CodePointMapData8<LineBreak>`, with support for `wordBreak`, `lineBreak`, and `locale` options
- `assets/icu_capi.wasm` (~96 KB): custom-built ICU4X WASM with link-time symbol pruning, retaining only what line breaking requires
- `assets/data.postcard` (~348 KB): Unicode data blob with LSTM model, enabling word-level line breaking for SA-class scripts (Thai, Burmese, Khmer)
- `assets/data_simple.postcard` (~29 KB): simplified data blob without LSTM, suitable for server-side Node.js environments

See `packages/icu4satori/README.md` for build details.

#### playground (6 files)

| File | Change |
|------|--------|
| `playground/package.json` | Add `icu4satori: workspace:*` dependency |
| `playground/utils/twemoji.ts` | Add `notoColor` emoji provider (pointing to the Google Noto Emoji GitHub main branch); the existing providers (twemoji, fluent, fluentFlat, etc.) no longer cover Unicode 17, causing U16/U17 emoji in the `§10 Emoji` tab to fail to render |
| `playground/pages/index.tsx` | Add ICU4X toggle checkbox, `loadTextEngine()` loading logic, and Noto font support |
| `playground/public/icu_capi.wasm` | ICU4X WASM binary for the browser-side playground |
| `playground/public/data.postcard` | Unicode data blob for the browser-side playground |
| `playground/cards/playground-data.ts` | Add 15 `§`-prefixed tabs covering all affected CSS properties and script types |

## Usage example

```ts
import { init, createTextEngine } from 'icu4satori'

// Load WASM and data blob once at startup
await init(wasmBuffer)
const textEngine = createTextEngine(new Uint8Array(dataBlob))

const svg = await satori(<div>สวัสดีชาวโลก</div>, {
  width: 600,
  height: 400,
  fonts: [...],
  textEngine, // omit to fall back to the existing behavior
})
```

## Testing

15 interactive playground tabs (`§1.1` through `§12`) are included, covering:
- Thai/SA-class script LSTM word breaking (`§1.1`)
- `word-break: break-all / keep-all / break-word` (`§1.2–§1.4`)
- `line-break: loose / normal / strict / anywhere` (`§4`)
- Emoji ZWJ sequences, Unicode 15–17 skin tone modifiers (`§10`)
- NBSP, empty string, pure whitespace edge cases (`§11`, `§12`)

Each tab has an **ICU4X TextEngine** toggle checkbox so you can flip between the two engines and compare output side by side.

**There are no unit tests yet.** I wasn't sure what the right approach would be — see Open Questions below.

#### 1. Testing strategy

The existing tests in `test/` are screenshot-based and require fonts and a full render pipeline. Loading ICU4X WASM asynchronously in that environment needs some setup I wasn't sure how to wire up correctly.

I can go a few directions — add a dedicated screenshot test suite for the `textEngine` path in CI, or unit-test `splitByBreakOpportunities()` with a mocked `TextEngine` to keep things simpler. Happy to follow whatever approach you prefer, or to add tests however you'd like me to structure them.

#### 2. Should the `§` playground tabs be kept?

These tabs serve as living documentation for the differences between ICU4X and the default engine. That said, they're noise for anyone not using ICU4X, and I'm not sure they belong in the main playground long-term. I'm happy to move them to a separate section, put them behind a flag, or remove them entirely — just let me know what feels right.

#### 3. Should the WASM and data files be committed to the repo?

**Background**: the intended deployment pattern for `icu4satori` is to load the WASM and data blob from a CDN URL at runtime (e.g. `init('https://cdn.example.com/icu_capi.wasm')`), rather than bundling them. The playground puts these files in `public/` and loads them via `window.location.origin` to faithfully reproduce that CDN loading path in the browser — so what reviewers see in the playground matches real Edge Function behavior.

Currently both `playground/public/icu_capi.wasm` (~96 KB) and `playground/public/data.postcard` (~348 KB) are committed directly to the repo. I'm open to downloading them via a postinstall script instead, or referencing them from `icu4satori`'s subpath exports to avoid duplication — whatever you think fits better.

#### 4. `lineBreak` inheritance fix

I added `lineBreak` to the inheritable CSS properties in `src/handler/inheritable.ts`. This is the correct CSS behavior (`line-break` is an inherited property per spec), but it's technically a behavior change — edge cases where a parent has `line-break` set and a child was previously not inheriting it could be affected. Should this be called out separately in the changelog, or is it fine to include as-is?

#### 5. `satori/full` entry point

You mentioned `satori/full` in #687 as a possible home for heavier dependencies. I went with a separate `icu4satori` package instead, mainly because the data blob needs to be loaded on demand at runtime (not bundled) and users need to be able to choose between the LSTM and simple variants. That said, if a `satori/full` entry point better fits the long-term plan I'm happy to revisit the packaging — let me know what you'd prefer.

#### 6. Where should `icu4satori` live?

Right now it's a subpackage inside the satori monorepo (`packages/icu4satori/`). I noticed that `@shuding/opentype.js` — a similar satori-adjacent package — is maintained as a standalone repo rather than a monorepo subpackage. I wasn't sure which approach you'd prefer here: keeping it in-tree makes atomic commits and CI easier, but a standalone repo gives it an independent release cadence. Happy to move it out if that's a better fit.

## Not included in this PR

- **README update**: deferred pending resolution of Open Questions 1, 5, and 6 — the final API surface, entry point, and package name may change. Happy to add documentation once those are settled.
- **Phase 2–6**: word/grapheme segmentation, Unicode property APIs, script detection, case mapping, BiDi — to be addressed in follow-up PRs as described in the tracking issue.

### CODE REVIEW

File: packages/icu4satori/package.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "icu4satori",
+  "version": "0.0.1",
+  "description": "ICU4X-powered text engine for satori",
+  "type": "module",
```
[the remark this answers is not in the document]
Comment (Vizards) [reply packages/icu4satori/package.json:5]: The Vercel deployment is failing because icu4satori's `dist` is gitignored and isn't built before the playground's next build runs — so import 'icu4satori' can't resolve the module.

Before I push a fix, could you help me understand how the Vercel project is configured? I can't find the `vercel.json` @shuding 

1. Build command: Is it set to something like turbo run build, or does Vercel go directly into the playground?
2. Root directory: Is the Vercel project root set to the repo root, or to playground?

Knowing this will tell me whether the right fix is a turbo pipeline (turbo.json), a vercel.json with a custom build command, or just prepending the build step in package.json

**Comment by Vizards** [comment]
`dist` is committed because the build requires the diplomat-generated `vendor/` bindings, which in turn require a Rust toolchain that hosted CI environments (Vercel, standard GitHub Actions) don't provide. 

A proper CI workflow (installing Rust to regenerate both) can be added as a follow-up. But it depends on the question: Where should icu4satori live
